futures/tests/io_buf_writer.rs

```diff
@@ -0,0 +1,247 @@
+use futures::executor::block_on;
+use futures::future::Future;
+use futures::io::{AsyncSeek, AsyncSeekExt, AsyncWrite, AsyncWriteExt, BufWriter, SeekFrom};
+use futures::task::{Context, Poll};
+use futures_test::task::noop_context;
+use std::io::{self, Cursor};
+use std::pin::Pin;
+
+macro_rules! run_write {
```

Comment:
Is there a reason for this rather than `writer.write(&mut buf).await`?

Reply:
We don't have `AsyncWriteExt::write` yet (but I think it makes sense to have it).

Reply:
Opened https://github.com/rust-lang-nursery/futures-rs/pull/1612.

Reply:
Merged

Reply:
Thanks!